pkg/pxc/app/statefulset/node_test.go

```diff
@@ -47,35 +140,40 @@ func TestSidecarContainers_PXCNode(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			cr := &api.PerconaXtraDBCluster{
 				ObjectMeta: metav1.ObjectMeta{
-					Name: "test-cluster",
+					Name:      "test-cluster",
+					Namespace: "test-ns",
 				},
 				Spec: api.PerconaXtraDBClusterSpec{
-					CRVersion: tt.crVersion,
-					PXC: &api.PXCSpec{
-						PodSpec: &tt.spec,
-					},
+					CRVersion: version.Version(),
+					PXC:       &tt.spec,
 				},
 			}
 
+			objs := []runtime.Object{cr}
+			if tt.envFromSecret != nil {
+				objs = append(objs, tt.envFromSecret)
+			}
+
+			client := test.BuildFakeClient(objs...)
+
 			pxcNode := Node{cr: cr}
 
-			c, err := pxcNode.AppContainer(&tt.spec, tt.secrets, cr, nil)
+			c, err := pxcNode.AppContainer(t.Context(), client, tt.spec.PodSpec, tt.secrets, cr, nil)
 			if tt.expectError {
-				if err == nil {
-					t.Errorf("expected error, got nil")
-				}
-				return
+				require.Error(t, err)
```

Comment:
The function `AppContainer` never returns an error af far as I can see. Is there a point in asserting an error?